src/cli/cli.cpp

```diff
@@ -554,6 +595,7 @@ void Interpreter::ProcessCounters(int argc, char *argv[])
             sServer->OutputFormat("    RxErrSec: %d\r\n", counters->mRxErrSec);
             sServer->OutputFormat("    RxErrFcs: %d\r\n", counters->mRxErrFcs);
             sServer->OutputFormat("    RxErrOther: %d\r\n", counters->mRxErrOther);
+            otFreeMemory(counters);
```

Comment:
Just noticed that `otGetMacCounters()` on Windows returns dynamically allocated memory that the caller needs to free.   However, the implementation in `src/core/openthread.cpp` does not behave this way.  Is there a way to avoid dynamically allocated memory with the Windows implementation?  Alternatively, we could change `otGetMacCounters()` to pass the `otMacCounters` struct and do a copy.

I also noticed a number of similar instances below.